sdk/storage/ci.yml

```diff
@@ -33,7 +33,13 @@ extends:
       LiveTestCtestRegex: azure-storage
       Clouds: Preview
       SupportedClouds: Preview
-      UseFederatedAuth: false
+      UseFederatedAuth: true
+      CloudConfig:
+        Preview:
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources-preview)
+          ServiceConnection: azure-sdk-tests
+          SubscriptionConfigurationFilePaths:
+            - eng/common/TestResources/sub-config/AzurePublicMsft.json
```

Comment:
@weshaggard I am not sure who's the best person to review changes like this, from engsys.

What required adding these, specifically for storage?
This archetype has it:
https://github.com/Azure/azure-sdk-for-cpp/blob/21b4db68f637cd6036535e671a256060c7446cfa/eng/pipelines/templates/stages/archetype-sdk-client.yml#L58-L61

Reply:
Things still work without these lines. I don't know what they do. I just copy-paste from elsewhere.

Reply:
you should only need the UseFederated line , Daniel made all the other required changes in the eng folder , i don't think the lines have any adverse effect since the pipelined passed , but i would still remove them to prevent accidental side effects since it's in the common code

Reply:
> you should only need the UseFederated line , Daniel made all the other required changes in the eng folder , i don't think the lines have any adverse effect since the pipelined passed , but i would still remove them to prevent accidental side effects since it's in the common code

@Jinming-Hu based on George's comment, can you update your PR to only have the 
`UseFederatedAuth: true` line

Reply:
Correct you shouldn't need this here however we do need to make a fix for the common on which I just submitted a PR for https://github.com/Azure/azure-sdk-for-cpp/pull/5826.